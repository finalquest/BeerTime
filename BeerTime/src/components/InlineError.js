import React from 'react';
import {
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';
import Label from './Label';
import { StylePropType } from '../utils/PropTypes';

const styles = StyleSheet.create({
  error: {
    fontSize: 11,
    color: 'red',
  },
});

const InlineError = ({ error, style }) => (error !== '' ? <Label style={[styles.error, style]}>{error}</Label> : null);

InlineError.propTypes = {
  error: PropTypes.string.isRequired,
  style: StylePropType,
};

InlineError.defaultProps = {
  style: {},
};

export default InlineError;
