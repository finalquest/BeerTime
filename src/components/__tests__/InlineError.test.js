import React from 'react';
import { shallow } from 'enzyme';

import InlineError from '../InlineError';

describe('InlineError test suite', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<InlineError error="pepe" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Label')).toHaveLength(1);
  });

  it('renders as expected with no error', () => {
    const wrapper = shallow(<InlineError error="" />);
    expect(wrapper.find('Label')).toHaveLength(0);
  });
});
