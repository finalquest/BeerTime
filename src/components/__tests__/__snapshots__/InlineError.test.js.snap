// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineError test suite renders as expected 1`] = `
<Label
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "red",
        "fontSize": 11,
      },
      Object {},
    ]
  }
  uppercase={false}
>
  pepe
</Label>
`;
